cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/TagValue.java

```diff
@@ -23,7 +23,7 @@
 /**
  * Represents tag and its value assigned to the {@link Fact}.
  */
-public final class TagValue {
+public final class TagValue implements Comparable<TagValue> {
```

Comment:
Its hard to understand what does it mean to compare TagValues. I'd rather pass Comparator into the SortedSet when you create it above